### PR TITLE
[BIOMAGE-1740] - Only show available columns in advanced option filters

### PR DESCRIPTION
**Pull request by aerlaut**
# Description
### Background
The advanced filtering menu available after running DE uses columns available in the DE table. Right now, you can choose to filter using a column that is not available in the DE table. This PR updates this behaviour only columns available in the DE table can be used for advanced filtering.

### To test
All the tests below are run using Vicky's multisample murine dataset

**To see all options**
- Go to Data Exploration and open the Differential Expression tab
- Choose "Compare cell set within a sample / group"
- Select "Cluster 0" for the 1st option
- Select "All other cells" for the 2nd option
- Select "All" for the 3rd option
- Run Differential Expression

Once the DE result is available, click "Advanced Filtering". Filtering over the preset filters, you should be able to see 3 preset filters. Click "Add custom filter" and click "Select Property" dropdown. You should be able to see all the columns available as options.

**To see limited options**
- Go to Data Exploration and open the Differential Expression tab
- Choose "Compare a selected cell set between samples/groups"
- Select "Cluster 14" for the 1st option
- Select "KO" for the 2nd option
- Select "WT1" for the 3rd option
- Run Differential Expression

Once the DE result is available, you should see that there is ony 1 column available in the DE table: logFC. 

Open "Advanced Filtering". Hovering over the preset filters, you should be able to see 2 preset filters: `Up-regulated` and `Down-regulated`. The 3rd option, `Significant`, is not avaiable because it requires the `p_val_adj` column. Click "Add custom filter" and click "Select Property" dropdown. You should be able to see only the `logFC` column, because the other columns are not available for selection.

# Details
#### URL to issue
https://biomage.atlassian.net/browse/BIOMAGE-1740

#### Link to staging deployment URL (or set N/A)
https://ui-agi-651.scp-staging.biomage.net/

#### Links to any PRs or resources related to this PR
N/A

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [X] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [X] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [X] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [x] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [x] Started end-to-end tests on the latest commit.

### Documentation updates
- [X] Relevant Github READMEs updated **or** no GitHub README updates required.
- [X] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

File: src/components/data-exploration/differential-expression-tool/AdvancedFilteringModal.jsx [hunk 1]
```diff
@@ -31,40 +31,67 @@ const valueRestrictions = {
...
+
+    const filteredCriteriaOptions = criteriaOptions
+      .filter((option) => availableColumns.includes(option.value));
+
+    console.log('*** filteredPresetFilters', filteredPresetFilters);
+    console.log('*** filteredCriteriaOptions', filteredCriteriaOptions);
+
+    setAvailablePresetFilters(filteredPresetFilters);
+    setAvailableCriteriaOptions(filteredCriteriaOptions);
+  }, [diffExprData.length]);
```
Comment (StefanBabukov) [inline src/components/data-exploration/differential-expression-tool/AdvancedFilteringModal.jsx]: What if the diffexprdata changes but the length is the same, will this retrigger?

File: src/components/data-exploration/differential-expression-tool/AdvancedFilteringModal.jsx [hunk 2]
```diff
@@ -31,40 +31,67 @@ const valueRestrictions = {
...
+    const availableColumns = Object.keys(diffExprData[0]);
+
+    const filteredPresetFilters = presetFilters
+      .filter((filter) => availableColumns.includes(filter.columnName));
+
+    const filteredCriteriaOptions = criteriaOptions
+      .filter((option) => availableColumns.includes(option.value));
+
+    console.log('*** filteredPresetFilters', filteredPresetFilters);
+    console.log('*** filteredCriteriaOptions', filteredCriteriaOptions);
```
Comment (StefanBabukov) [inline src/components/data-exploration/differential-expression-tool/AdvancedFilteringModal.jsx]: remove logs

**Comment by StefanBabukov** [comment]
Add a test to check if correct columns are used for the filters  and criteria

File: src/components/data-exploration/differential-expression-tool/AdvancedFilteringModal.jsx (hunk 1 above)
Comment (aerlaut) [reply src/components/data-exploration/differential-expression-tool/AdvancedFilteringModal.jsx]: Oh right, this won't retrigger. This should retrigger if the number of column changes. I'll update this.

File: src/components/data-exploration/differential-expression-tool/AdvancedFilteringModal.jsx (hunk 2 above)
Comment (aerlaut) [reply src/components/data-exploration/differential-expression-tool/AdvancedFilteringModal.jsx]: removed

**Review by StefanBabukov — APPROVED** [review]
nice tests lgtm

**Pull request merged**